test/with_dummyserver/test_socketlevel.py

```diff
@@ -521,6 +521,47 @@ def socket_handler(listener):
         finally:
             timed_out.set()
 
+    def test_ssl_failed_fingerprint_verification(self):
+        def socket_handler(listener):
+            for i in range(2):
+                sock = listener.accept()[0]
+                ssl_sock = ssl.wrap_socket(sock,
+                                           server_side=True,
+                                           keyfile=DEFAULT_CERTS['keyfile'],
+                                           certfile=DEFAULT_CERTS['certfile'],
+                                           ca_certs=DEFAULT_CA)
+
+                buf = b''
+                while not buf.endswith(b'\r\n\r\n'):
+                    buf += ssl_sock.recv(65536)
```

Comment:
This buffer does not get any data while running this test. I am not sure why. I assume fixing this will also fix the test.